packages/react-tokens/src/variablesByFile.js

```diff
@@ -0,0 +1,269 @@
+const glob = require('glob');
+const { dirname, resolve, join, basename } = require('path');
+const { parse } = require('css');
+const { readFileSync, readdirSync } = require('fs');
+const { outputFileSync } = require('fs-extra');
+
+const outDir = resolve(__dirname, '../dist/variables');
+const pfStylesDir = dirname(require.resolve('@patternfly/patternfly/patternfly.css'));
+const templateDir = resolve(__dirname, './templates');
+
+const cssFiles = glob.sync('{**/{components,layouts}/**/*.css,**/patternfly-charts.css,**/patternfly-variables.css}', {
+  cwd: pfStylesDir,
+  ignore: ['assets/**']
+});
+
+const formatCustomPropertyName = key => key.replace('--pf-', '').replace(/-+/g, '_');
+
+// various lookup tables to resolve variables
+
+const varsRegex = /(--.*):\s*#{(\$pf-[^\s]+)}/g;
+const variables = readFileSync(require.resolve('@patternfly/patternfly/_variables.scss'), 'utf8');
+let matches;
+const cssGlobalsToScssVarsMap = {};
+while ((matches = varsRegex.exec(variables))) {
+  cssGlobalsToScssVarsMap[matches[1]] = matches[2];
+}
+
+const scssVarsRegex = /(\$.*):\s*([^;^!]+)/g;
+// contains default values and mappings to colors.scss for color values
+const scssVariables = readFileSync(
+  require.resolve('@patternfly/patternfly/sass-utilities/scss-variables.scss'),
+  'utf8'
+);
+matches = null;
+const scssVarsMap = {};
+while ((matches = scssVarsRegex.exec(scssVariables))) {
```

Comment:
=== in `while` loop, can also remove extra parens ( )

Reply:
it assigns the result to matches and then implicitly checks if matches is not null or undefined